src/passes/ifStatementTempVarPostpender.ts

```diff
@@ -0,0 +1,200 @@
+  }
+
+  visitBlock(node: Block, ast: AST): void {
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitUncheckedBlock(node: UncheckedBlock, ast: AST): void {
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitVariableDeclarationStatement(node: VariableDeclarationStatement, ast: AST): void {
+    node.vDeclarations.forEach((declaration) => this.liveVars.delete(declaration.name));
+    if (node.vInitialValue) {
+      this.dispatchVisit(node.vInitialValue, ast);
+    }
+  }
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    // ignore the function identifier
+    for (const arg of node.vArguments) {
+      this.dispatchVisit(arg, ast);
+    }
+  }
+
+  visitIfStatement(node: IfStatement, ast: AST): void {
+    ensureBothBranchesAreBlocks(node, ast);
+
+    const liveVarsBefore = new Set(this.liveVars);
+    this.dispatchVisit(node.vTrueBody, ast);
+    const liveVarsAfterTrueBody = new Set(this.liveVars);
+    this.liveVars = new Set(liveVarsBefore);
+    let liveVarsAfterFalseBody = new Set<string>();
+    if (node.vFalseBody) {
+      this.dispatchVisit(node.vFalseBody, ast);
+      liveVarsAfterFalseBody = new Set(this.liveVars);
+    }
+    this.liveVars = new Set(liveVarsBefore);
+    this.dispatchVisit(node.vCondition, ast);
+    this.liveVars = new Set([
+      ...this.liveVars,
+      ...liveVarsAfterFalseBody,
+      ...liveVarsAfterTrueBody,
+    ]);
+    const trueHasPathWithoutReturn = hasPathWithoutReturn(node.vTrueBody);
+    if (trueHasPathWithoutReturn) {
+      for (const liveVar of liveVarsBefore) {
+        const child = createCairoTempVar(liveVar, ast);
+        (node.vTrueBody as Block).appendChild(child);
+        ast.registerChild(child, node.vTrueBody);
+      }
+    }
+    const falseHasPathWithoutReturn =
+      (node.vFalseBody && hasPathWithoutReturn(node.vFalseBody)) || !node.vFalseBody;
+    if (falseHasPathWithoutReturn) {
+      for (const liveVar of liveVarsBefore) {
+        const child = createCairoTempVar(liveVar, ast);
+        (node.vFalseBody as Block).appendChild(child);
+        // @ts-ignore ensureBothBranchesAreBlocks forces vFalseBody to be defined;
+        ast.registerChild(child, node.vFalseBody);
+      }
+    }
+
+    if (falseHasPathWithoutReturn || trueHasPathWithoutReturn) {
+      const parent = node.parent;
+      const blockWithAfter = createBlock(
+        [node, ...[...liveVarsBefore].map((v) => createCairoTempVar(v, ast))],
+        ast,
+      );
+      ast.replaceNode(node, blockWithAfter, parent);
+    }
+  }
+}
+
+function ensureBothBranchesAreBlocks(node: IfStatement, ast: AST): void {
+  if (!(node.vTrueBody instanceof Block) && !(node.vTrueBody instanceof UncheckedBlock)) {
+    node.vTrueBody = createBlock([node.vTrueBody], ast);
+    ast.registerChild(node.vTrueBody, node);
+  }
+
+  if (
+    node.vFalseBody &&
```

Comment:
opinionated nitpick: don't leave to JavaScript engine to coerce undefined to boolean or vice versa, instead declare it explicitly like `node.vFalseBody !== undefined`